2004-10-05  Richard Henderson  <rth@redhat.com>

	* config/ia64/ia64.c (ia64_expand_load_address): Recurse for
	base of addition.

--- gcc/config/ia64/ia64.c	2003/11/21 14:18:53	1.139.2.15.8.11
+++ gcc/config/ia64/ia64.c	2004/10/05 22:24:12	1.139.2.15.8.12
@@ -1052,8 +1052,7 @@
       if (! scratch)
 	scratch = no_new_pseudos ? subtarget : gen_reg_rtx (DImode);
 
-      emit_insn (gen_load_symptr (subtarget, plus_constant (sym, hi),
-				  scratch));
+      ia64_expand_load_address (subtarget, plus_constant (sym, hi), scratch);
       emit_insn (gen_adddi3 (temp, subtarget, GEN_INT (lo)));
     }
   else
